server/src/main/java/io/druid/offheap/OffheapBufferPool.java

```diff
@@ -47,7 +47,8 @@ public ByteBuffer get()
             );
             return ByteBuffer.allocateDirect(computationBufferSize);
           }
-        }
+        },
+        cacheSize
```

Comment:
There's no guard here to ensure we don't overflow direct memory limits of the JVM. is that intended?